editor/plugins/spatial_editor_plugin.cpp

```diff
@@ -3935,6 +3947,48 @@ void SpatialEditor::_menu_item_pressed(int p_option) {
 
 			settings_dialog->popup_centered(settings_vbc->get_combined_minimum_size() + Size2(50, 50));
 		} break;
+		case MENU_LOCK_SELECTED: {
+
+			List<Node *> &selection = editor_selection->get_selected_node_list();
+
+			for (List<Node *>::Element *E = selection.front(); E; E = E->next()) {
+
+				Spatial *spatial = Object::cast_to<Spatial>(E->get());
+				if (!spatial || !spatial->is_visible_in_tree())
+					continue;
+
+				if (spatial->get_viewport() != EditorNode::get_singleton()->get_scene_root())
+					continue;
+
+				//SpatialEditorSelectedItem spatial_data = editor_selection.get_node_editor_data<SpatialEditorSelectedItem>(spatial);
+				spatial->set_meta("_edit_lock_", true);
+				emit_signal("item_lock_status_changed");
+			}
+			//viewport->update();
+
+			_refresh_menu_icons();
+		} break;
+		case MENU_UNLOCK_SELECTED: {
+
+			List<Node *> &selection = editor_selection->get_selected_node_list();
+
+			for (List<Node *>::Element *E = selection.front(); E; E = E->next()) {
+
+				Spatial *spatial = Object::cast_to<Spatial>(E->get());
+				if (!spatial || !spatial->is_visible_in_tree())
+					continue;
+
+				if (spatial->get_viewport() != EditorNode::get_singleton()->get_scene_root())
+					continue;
+
+				//SpatialEditorSelectedItem spatial_data = editor_selection.get_node_editor_data<SpatialEditorSelectedItem>(spatial);
+				spatial->set_meta("_edit_lock_", Variant());
+				emit_signal("item_lock_status_changed");
+				//emit_signal("item_lock_status_changed");
```

Comment:
Doesn't seem really useful to keep? ;)
Please check the other commented statements and see if they're needed or should be removed.

Reply:
oh, my bad will clean them asap